internal/bminventory/inventory.go

```diff
@@ -1254,6 +1255,76 @@ func (b *bareMetalInventory) InstallClusterInternal(ctx context.Context, params
 	return cluster, nil
 }
 
+func (b *bareMetalInventory) InstallSingleDay2HostInternal(ctx context.Context, clusterId strfmt.UUID, hostId strfmt.UUID) error {
+
+	log := logutil.FromContext(ctx, b.log)
+	var err error
+	var cluster *common.Cluster
+	var h *models.Host
+
+	if cluster, err = common.GetClusterFromDB(b.db, clusterId, common.UseEagerLoading); err != nil {
+		return err
+	}
+	if h, err = b.getHost(ctx, clusterId.String(), hostId.String()); err != nil {
+		return err
+	}
+	// auto select host roles if not selected yet.
+	err = b.db.Transaction(func(tx *gorm.DB) error {
+		if err = b.hostApi.AutoAssignRole(ctx, h, tx); err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if err = b.refreshAllHosts(ctx, cluster); err != nil {
```

Comment:
why do you need to call refresh all hosts? maybe just the relevant host?

Reply:
maybe we can remove transaction from it? it should set only "worker role" in auto assign part. Does it worth all that logic?

Reply:
I refreshed only the specific host.